subcommand/controller/command.go

```diff
@@ -107,22 +111,31 @@ func (c *Command) Run(args []string) int {
 	}
 
 	var zapLevel zapcore.Level
+	// It is possible that a user passes in "trace" from global.logLevel, until we standardize on one logging framework
+	// we will assume they meant debug here and not fail.
+	if c.flagLogLevel == "trace" || c.flagLogLevel == "TRACE" {
+		c.flagLogLevel = "debug"
+	}
 	if err := zapLevel.UnmarshalText([]byte(c.flagLogLevel)); err != nil {
 		c.UI.Error(fmt.Sprintf("Error parsing -log-level %q: %s", c.flagLogLevel, err.Error()))
 		return 1
 	}
-	// We set UseDevMode to true because we don't want our logs json
-	// formatted.
-	logger := zap.New(zap.UseDevMode(true), zap.Level(zapLevel))
-	ctrl.SetLogger(logger)
-	klog.SetLogger(logger)
+
+	var zapLogger logr.Logger
+	if c.flagLogJson {
+		zapLogger = zap.New(zap.UseDevMode(false), zap.Level(zapLevel), zap.JSONEncoder())
+	} else {
+		zapLogger = zap.New(zap.UseDevMode(false), zap.Level(zapLevel), zap.ConsoleEncoder())
+	}
```

Comment:
Can we move this to a function in `common`: `func ZapLogger(level string, jsonLogging bool) (logr.Logger, error)`

And use the same function in `inject-connect`.

Reply:
great idea, done!